### PR TITLE
feat: add coingecko api key

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Use Coingecko's api key in the proxy requests

### CODE REVIEW

File: src/routes/coingeckoProxy/index.ts [hunk 1]
```diff
@@ -9,9 +9,13 @@ const coingeckoProxy: FastifyPluginAsync = async (
   opts
 ): Promise<void> => {
   fastify.register(httpProxy, {
-    upstream: "https://api.coingecko.com",
-    rewritePrefix: "/api/v3/simple/",
+    upstream: "https://pro-api.coingecko.com",
```
Comment (alfetopito) [inline src/routes/coingeckoProxy/index.ts:12]: Our key has been temporarily set to "pro" for testing.
For this reason, we must use the pro api endpoint.

I'll switch this to and env var so we don't need a deployment to change it.

File: src/routes/coingeckoProxy/index.ts [hunk 2]
```diff
@@ -3,15 +3,22 @@ import { FastifyPluginAsync } from "fastify";
...
 const coingeckoProxy: FastifyPluginAsync = async (
   fastify,
   opts
 ): Promise<void> => {
   fastify.register(httpProxy, {
-    upstream: "https://api.coingecko.com",
-    rewritePrefix: "/api/v3/simple/",
+    upstream:
+      fastify.config.COINGECKO_PROXY_UPSTREAM ||
+      DEFAULT_COINGECKO_PROXY_UPSTREAM,
```
Comment (alfetopito) [inline src/routes/coingeckoProxy/index.ts:15]: Use the env var `COINGECKO_PROXY_UPSTREAM` when set

File: src/routes/coingeckoProxy/index.ts [hunk 3]
```diff
@@ -3,15 +3,22 @@ import { FastifyPluginAsync } from "fastify";
...
+    rewritePrefix: "/api/v3/",
     replyOptions: {
+      rewriteRequestHeaders: (request, headers) => ({
+        ...headers,
+        "x-cg-pro-api-key": fastify.config.COINGECKO_API_KEY,
```
Comment (alfetopito) [inline src/routes/coingeckoProxy/index.ts:20]: Add required env var `COINGECKO_API_KEY` to requests

**Review by anxolin — COMMENTED** [review]
Approve! thanks

**Pull request merged**